src/requires/spglib.jl

```diff
@@ -0,0 +1,46 @@
+# Functionalities that are available when Spglib is loaded
+
+using Bravais: reciprocalbasis, DirectBasis
+using LinearAlgebra
+import ..KPaths: irrfbz_path
+
+export irrfbz_path
+
+"""
+    irrfbz_path(cell::Spglib.Cell)  -->  ::KPath{D}
+
+Returns a k-path for arbitrary 3D cell in the Spglib format. The k-path is equivalent to the
+"standard" k-path given by `irrfbz_path` for the equivalent standard primitive lattice.
+Does not give the correct k-path if the input cell is a supercell of a smaller primitive
+cell (a warning is printed).
+"""
+function irrfbz_path(cell::Spglib.Cell)
+    # standardize cell
+    dset = Spglib.get_dataset(cell)
+    sgnum = dset.spacegroup_number
+    std_lattice = DirectBasis(collect(eachcol(dset.std_lattice)))
+
+    # If the input cell is a supercell (without any distortion), then the irrfbz algorithm cannot work.
+    # Check the volume of the input cell and primitive cell are equal.
+    if round(Int, det(cell.lattice) / det(dset.primitive_lattice)) != 1
+        @warn "input cell is a supercell. irrfbz Does not give a correct k path."
```

Comment:
```suggestion
        errror(DomainError(cell, "`cell` is a supercell and untreatable by `irrfbz_path`.")
```

I think it is better to just throw an error here, right? There's no point in returning anything if we know it's going to be wrong somehow anyway.

Additionally, would this also fail if `cell` represents a conventional lattice? In that case, maybe we should also note that option in the error message.

Reply:
> I think it is better to just throw an error here, right? There's no point in returning anything if we know it's going to be wrong somehow anyway.

Yes, I agree.

> Additionally, would this also fail if cell represents a conventional lattice? In that case, maybe we should also note that option in the error message.

Yes, if the conventional lattice is a supercell. (e.g. cubic conventional lattice of a fcc primitive lattice).
Would "If the cell represents a standard conventional lattice, use `irrfbz_path(sgnum::Integer, Rs)`." be okay?

Reply:
Also I assume `errror` is a typo of `error` 😄